rs/nns/governance/src/neuron/types.rs

```diff
@@ -348,14 +353,57 @@ impl Neuron {
             > 0
     }
 
+    fn deciding_voting_power_adjustment_factor(
+        duration_since_voting_power_refreshed: Duration,
+    ) -> Decimal {
+        let linear_map = LinearMap::new(
+            Decimal::from(6 * ONE_MONTH_SECONDS)..Decimal::from(7 * ONE_MONTH_SECONDS), // from
+            Decimal::from(1)..Decimal::from(0),                                         // to
+        );
+
+        linear_map
+            .apply(Decimal::from(
+                duration_since_voting_power_refreshed.as_secs(),
+            ))
+            .clamp(Decimal::from(0), Decimal::from(1))
+    }
+
+    /// How much swap this neuron has when it casts its vote on proposals.
+    pub fn deciding_voting_power(&self, now_seconds: u64) -> u64 {
+        // Main inputs.
+        let adjustment_factor: Decimal = if is_voting_power_adjustment_enabled() {
+            Self::deciding_voting_power_adjustment_factor(Duration::from_secs(
+                now_seconds.saturating_sub(self.voting_power_refreshed_timestamp_seconds),
+            ))
+        } else {
+            Decimal::from(1)
+        };
+        let potential_voting_power = self.potential_voting_power(now_seconds);
+
+        // Main calculation.
+        let result = adjustment_factor * Decimal::from(potential_voting_power);
+
+        // Convert (back) to u64.
+        let result = result.round();
```

Comment:
Please double check which type of rounding is most appropriate in this case, see different rounding strategies described [here](https://docs.rs/rust_decimal/latest/rust_decimal/enum.RoundingStrategy.html).

Reply:
Thanks. I didn't know about all of these options.

I think it doesn't matter which one is used.

Nevertheless, there should be a comment that explicitly says so (and explains why that is).